### PR TITLE
BLOCKCHAIN-129 - apply for congressman feature

### PR DESCRIPTION
**Pull request by MichalLustykNeti**

### CODE REVIEW

**Review by kacperzuk-neti — CHANGES_REQUESTED** [review]

File: src/api/nodeRpcCall.js [hunk 1]
```diff
@@ -1298,6 +1298,35 @@ const getBlockEvents = async (blockHash) => {
   }
 }
 
+const getCongressCandidates = async () => {
+  const api = await getApi();
+  const electionsCandidates = await api.query.elections.candidates()
+  return electionsCandidates.toHuman()
```
Comment (kacperzuk-neti) [inline src/api/nodeRpcCall.js:1304]: Not a blocker, but in the future only use `.toHuman()` when displaying stuff to a human. Definitely not in sagas/nodeRpcCall.js.

File: src/redux/sagas/congress.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  const { errorData } = yield cps(applyForCongress, walletAddress);
+  if (errorData.isError) {
+    yield put(
+      blockchainActions.setErrorExistsAndUnacknowledgedByUser.success(true),
+    );
+    yield put(blockchainActions.setError.success(errorData));
+    yield put(congressActions.applyForCongress.failure(errorData));
+  } else {
+    yield put(congressActions.applyForCongress.success());
```
Comment (kacperzuk-neti) [inline src/redux/sagas/congress.js:23]: We should update congress candidates on success.

File: src/redux/reducers/congress.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  {
+    [combineActions(
+      congressActions.applyForCongress.call,
+      congressActions.getCongressCandidates.call,
+    )]: (state) => ({
+      ...state,
+      loading: true,
+    }),
+    [combineActions(
+      congressActions.applyForCongress.success,
```
Comment (kacperzuk-neti) [inline src/redux/reducers/congress.js:19]: We shouldn't yet set loading to false on congressActions.applyForCongress.success.
On success, we should dispatch congressActions.getCongressCandidates.call to update list of candidates, and success of this action will set loading to false.

File: src/redux/sagas/index.js [hunk 4]
```diff
@@ -80,5 +81,9 @@ export default function* rootSaga() {
     validatorSagas.payoutWatcher(),
     validatorSagas.getPendingRewardsWatcher(),
     validatorSagas.getInfoWatcher(),
+
+    // CONGRESS
+    congressSagas.applyForCongressWatcher(),
+    congressSagas.getCongressCandidatesWatcher()
```
Comment (kacperzuk-neti) [inline src/redux/sagas/index.js:87]:
```suggestion
    congressSagas.getCongressCandidatesWatcher(),
```

**Review by kacperzuk-neti — CHANGES_REQUESTED** [review]

**Review by kacperzuk-neti — APPROVED** [review]

**Review by DorianSternVukotic — APPROVED** [review]

**Pull request merged**